crates/nu-command/src/filesystem/ls.rs

```diff
@@ -900,3 +874,21 @@ mod windows_helper {
         false
     }
 }
+
+#[allow(clippy::type_complexity)]
+fn read_dir(
+    f: PathBuf,
+) -> Result<
+    (
+        Option<PathBuf>,
```

Comment:
We don't have to follow the signature of `glob_from()`. I think just returning the iterator is cleaner.